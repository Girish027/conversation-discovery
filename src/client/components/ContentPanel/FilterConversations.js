import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Checkbox, TextField } from '@tfs/ui-components';
import { filterPopupStyle } from '../../styles/index';
import _ from 'lodash';
import 'styles/sidebar.scss';

export class FilterConversations extends Component {
  constructor(props) {
    super(props);
    this.props = props;

    this.toggleCheckbox = this.toggleCheckbox.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);

    this.state = {
      similarityCutOff: '',
      selected: { ...this.props.filteredItems }
    };

    this.filterItems = {
      similarityCutOff: 'similarityCutOff',
    };
  }

  handleInputChange(val, filterItem) {
    if (filterItem === this.filterItems.similarityCutOff) {
      this.setState({
        similarityCutOff: parseFloat(val)
      });
    }
  }

  toggleCheckbox(checked, value) {
    if (value === this.filterItems.similarityCutOff) {
      let selected;
      if (checked && this.state.similarityCutOff !== '') {
        selected = { ...this.state.selected, similarity: this.state.similarityCutOff };
      } else {
        selected = { ...this.state.selected };
        delete (selected.similarity);
      }
      this.setState({
        selected
      }, this.props.onFilter(selected));
    }
  }

  render() {
    const { similarityCutOff, selected } = this.state;

    return (
      <div>
        <div style={filterPopupStyle.filterSimilarity}>
          <Checkbox
            key='one'
            value={this.filterItems.similarityCutOff}
            checked={!_.isEmpty(selected) && selected.similarity !== ''}
            label='Similarity Score greater than'
            onChange={(event) => {
              this.toggleCheckbox(event.target.checked, this.filterItems.similarityCutOff);
            }}
          />
          <TextField
            type='number'
            min={0}
            max={100}
            class='filter-input'
            styleOverride={filterPopupStyle.textboxStyle}
            placeholder={0.08}
            onChange={(event) => this.handleInputChange(event.target.value, this.filterItems.similarityCutOff)}
            name='similarity-score'
            value={selected.similarity || similarityCutOff}
          />
        </div>

        <div style={filterPopupStyle.filterSimilarity}>
          <Checkbox
            key='two'
            value=''
            label='Random Sample Size'
            onChange={(event) => {
              this.toggleCheckbox(event.target.checked, this.filterItems.similarityCutOff);
            }}
          />
          <TextField
            type='number'
            min={0}
            max={100}
            class='filter-input'
            styleOverride={filterPopupStyle.textboxStyle}
            placeholder={30}
            onChange={(event) => this.handleInputChange(event.target.value, this.filterItems.similarityCutOff)}
            name='sample-size'
            value={''}
          />
        </div>
      </div>
    );
  }
}

FilterConversations.propTypes = {
  onFilter: PropTypes.func,
  filteredItems: PropTypes.object
};

FilterConversations.defaultProps = {
  onFilter: () => {},
  filteredItems: {}
};

const mapStateToProps = () => {
};

const mapDispatchToProps = (dispatch) => ({
  dispatch,
});

export default connect(mapStateToProps, mapDispatchToProps)(FilterConversations);
