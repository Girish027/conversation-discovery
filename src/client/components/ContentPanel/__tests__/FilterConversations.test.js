import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { FilterConversations } from '../FilterConversations';

describe('<FilterConversations />', () => {
  let wrapper;

  const props = {};

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<FilterConversations
      {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    beforeEach(() => {
      props.onBlur = jest.fn();
      wrapper = getShallowWrapper(props);
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    describe('handleInputChange', () => {
      test('should change the state of similarity cutoff', () => {
        wrapper.instance().handleInputChange('0.21', 'similarityCutOff');
        expect(wrapper.state().similarityCutOff).toBe(0.21);
      });
    });

    describe('toggleCheckbox', () => {
      test('should change the state of selected value when similarity cut of is null', () => {
        wrapper.instance().toggleCheckbox(true, 'similarityCutOff');
        expect(wrapper.state().selected).toStrictEqual({});
      });

      test('should change the state of selected value when similarity cut of is not null', () => {
        wrapper.setState({similarityCutOff: 0.21});
        wrapper.instance().toggleCheckbox(true, 'similarityCutOff');
        expect(wrapper.state().selected).toEqual({similarity: 0.21});
      });
    });
  });
});
