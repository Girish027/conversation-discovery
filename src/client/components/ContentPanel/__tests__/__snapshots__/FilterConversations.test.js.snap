// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterConversations /> Snapshots renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "display": "flex",
        "paddingBottom": "5px",
      }
    }
  >
    <withTheme(Checkbox)
      checked={false}
      key="one"
      label="Similarity Score greater than"
      onChange={[Function]}
      value="similarityCutOff"
    />
    <withTheme(TextField)
      class="filter-input"
      max={100}
      min={0}
      name="similarity-score"
      onChange={[Function]}
      placeholder={0.08}
      styleOverride={
        Object {
          "input": Object {
            "height": "25px",
          },
          "marginLeft": "10px",
          "width": "50px",
        }
      }
      type="number"
      value=""
    />
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "paddingBottom": "5px",
      }
    }
  >
    <withTheme(Checkbox)
      key="two"
      label="Random Sample Size"
      onChange={[Function]}
      value=""
    />
    <withTheme(TextField)
      class="filter-input"
      max={100}
      min={0}
      name="sample-size"
      onChange={[Function]}
      placeholder={30}
      styleOverride={
        Object {
          "input": Object {
            "height": "25px",
          },
          "marginLeft": "10px",
          "width": "50px",
        }
      }
      type="number"
      value=""
    />
  </div>
</div>
`;
